src/pipelinedefinition.cpp

```diff
@@ -477,6 +477,21 @@ class NodeValidator {
         return StatusCode::OK;
     }
 
+    bool isConnectionShapeCorrect(shape_t tensorInputShape, shape_t tensorOutputShape) {
+        if (tensorInputShape.size() != tensorOutputShape.size()) {
+            return false;
+        }
+
+        for (size_t i = 0; i < tensorInputShape.size(); i++) {
+            if (tensorInputShape[i] != tensorOutputShape[i]) {
+                if (tensorInputShape[i] != 0 && tensorOutputShape[i] != 0) {
```

Comment:
I would merge these two if statements into one to not to create unnecessary nesting